### PR TITLE
Solve the rare pylint issues in the codebase

### PR DESCRIPTION
**Pull request by berland**
**Issue**

Partially solves #3309 by picking from the bottom of the list, the issues that we have the fewest of.

**Approach**
Look at the pylint error, perhaps do some googling, and look at the code. A single digit percentage of the pylint issues are real bugs.

This PR is organized with multiple commits, to make bisection possible in the case of bugs being introduced. It might make sense to do review commit by commit.

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: job_runner/reporting/event.py [hunk 1]
```diff
@@ -79,7 +81,7 @@ def _initialize_state_machine(self):
 
     def report(self, msg):
         new_state = None
-        for state in self._states.keys():
```
Comment (sondreso) [inline job_runner/reporting/event.py:82]: Does it not make sense to use keys() when we are just discarding the values anyway?

File: ert_gui/plottery/plot_style.py [hunk 2]
```diff
@@ -65,10 +66,8 @@ def alpha(self):
 
     @alpha.setter
     def alpha(self, alpha):
-        if alpha > 1.0:
-            alpha = 1.0
-        if alpha < 0.0:
-            alpha = 0.0
+        alpha = min(alpha, 1.0)
+        alpha = max(alpha, 0.0)
```
Comment (sondreso) [inline ert_gui/plottery/plot_style.py:70]: Very nitpick, but personally I think the clamp pattern can be written in one line like `alpha = max(min(alpha, 1.0), 0.0)`, but totally OK with leaving it as is

File: job_runner/reporting/event.py (hunk 1 above)
Comment (berland) [reply job_runner/reporting/event.py:82]: It does, distracted by the pylint message which says `items()`..

File: ert_gui/tools/manage_cases/case_init_configuration.py [hunk 3]
```diff
@@ -243,7 +243,7 @@ def _showInfoForCase(self, case_name=None):
         states = list(self.ert.getEnkfFsManager().getStateMapForCase(case_name))
 
         html = "<table>"
-        for index in range(len(states)):
+        for index, _ in enumerate(states):
```
[the remark this answers is not in the document]
Comment (berland) [reply ert_gui/tools/manage_cases/case_init_configuration.py:246]: No, we can do better! Fixed.

**Review by sondreso — APPROVED** [review]
LGTM!

**Pull request merged**